### ISSUE TITLE
Review correct way of importing Lodash libraries

### ISSUE THREAD

**Issue by asvinb** [google/site-kit-wp#6139]
## Bug Description

As part of #4610 , we are importing Lodash methods using different ways, named top level imports and direct imports. To have the smallest bundle size, direct imports should be preferred over named top level ones. We can either:

- [leverage the use of](https://www.npmjs.com/package/babel-plugin-lodash) `babel-plugin-lodash` which is in package.json but not actively used, together with `lodash-webpack-plugin` [as seen here](https://www.npmjs.com/package/lodash-webpack-plugin).
- or update named top lever imports to direct imports.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@asvinb @techanvil @tofumatt @kuasha420 feel free to draft AC here 👍

**Comment by tofumatt** [comment]
ACs here make sense to me, but it's worth investigating to see how well the `lodash-webpack-plugin` saves in bundle size vs developer intervention needed. This post https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/ and the [`lodash-webpack-plugin` README](https://github.com/lodash/lodash-webpack-plugin) both warn that lodash can behave in unexpected ways when that plugin is used.

Just using the Babel plugin might be good compromise between bundle size, developer experience, and limiting the possibility of bugs.

**Comment by derweili** [comment]
@tofumatt tested it and did not experience problems `lodash-webpack-plugin`. It saved 62kb on `googlesitekit-vendor`. So not much saving, althought I have to say that enabling the `babel-plugin-lodash` did not save anything, because we are already using the "tree-shaking-syntax".

But I'm not sure if we want to add a package to Site Kit thats itsef says "could cause problems".

**Comment by kuasha420** [comment]
@tofumatt We can partially use the webpack plugin, ie. disable things that we're definitely not using. The savings come from the webpack plugin, the babel plugin is only aesthetic and dx change for us at this point. What do you think?

@derweili Thank you for investigating and testing. I think your IB should also address the first point in AC and bump the estimate, once we agree with the AC fully that is.

**Comment by tofumatt** [comment]
Ah, I wasn't sure if we were already getting tree-shaking, seems like we are.

Reading the docs of the webpack plugin and seeing issues like this one: https://github.com/lodash/lodash-webpack-plugin/issues/177 makes me not want to integrate it into the plugin. It seems quite prone to breakage and might even break other modules without us noticing.

Let's keep this issue around cleaning up the imports only. I think the webpack plugin is too risky, even with the decent savings it comes with. I just don't think those 64kB are worth the potential headache 🙂

**Comment by kuasha420** [comment]
@tofumatt Agreed :100: percent with you after seeing the issues. Let's not do this. I've updated the AC. Cheers. cc @derweili

**Comment by derweili** [comment]
@kuasha420 there is a [eslint plugin `eslint-plugin-lodash`](https://github.com/wix/eslint-plugin-lodash)  which we could use to enforce the correct import syntax. I would not recommend to use one of their configurations, but we could use [this rule](https://github.com/wix/eslint-plugin-lodash/blob/master/docs/rules/import-scope.md) to enforce the "Destructured Members" import syntax.

So our eslint config would be extended by one rule:

```json
"lodash/import-scope": [
    "error",
]
```

What do you think about this?

**Comment by kuasha420** [comment]
@derweili That's a nice DX improvement, not sure if we should do it here or move this one forward as is and create a follow up issue to properly export the `[eslint-plugin-lodash](https://github.com/wix/eslint-plugin-lodash)`. I'd say go for the latter approach.

**Comment by derweili** [comment]
@kuasha420 ok great, will create a followup issue.

**Comment by eugene-manuilov** [comment]
> * Add `"lodash"` to plugin in `.babelrc`.

@derweili the `.babelrc` config isn't used by webpack, it is used by jest. We need to add the plugin in the webpack config itself: https://github.com/google/site-kit-wp/blob/1ac9c115cddc47053bcb2adff3c4034de232cc0a/webpack.config.js#L184-L190

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by techanvil** [comment]
No QA needed, moving to Approval.
